skimage/filters/rank/tests/test_rank.py

```diff
@@ -340,6 +380,16 @@ def test_compare_8bit_unsigned_vs_signed(self):
                 out_s = func(image_s, disk(3))
             assert_equal(out_u, out_s)
 
+        methods_3d = ['equalize', 'otsu']
+
+        for method in methods_3d:
+            func = getattr(rank, method)
+            out_u = func(volume_u, ball(3))
+            with expected_warnings(["Possible precision loss"]):
+                out_s = func(volume_s, ball(3))
+            assert_equal(out_u, out_s)
```

Comment:
Since both the setup and the test for 3D are separate, you might as well put this in a separate test method, e.g. `def test_compare_8bit_unsigned_vs_signed_3d(self):`

Reply:
fixed, did the same to test_compare_ubyte_vs_float_3d.